[Automation] native-image run fails for io.grpc:grpc-opentelemetry:1.80.0

## native-image run on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.80.0" ./gradlew clean test -Pcoordinates="io.grpc:grpc-opentelemetry:1.80.0"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25727047580/job/75542581240
Last 300 lines of the log:
```console
--add-exports=java.base/jdk.internal.module=org.graalvm.nativeimage.base,org.graalvm.nativeimage.builder \
--add-exports=java.base/jdk.internal.perf=org.graalvm.nativeimage.builder \
--add-exports=java.base/jdk.internal.platform=org.graalvm.nativeimage.builder \
--add-exports=java.base/jdk.internal.ref=org.graalvm.nativeimage.builder,org.graalvm.nativeimage.objectfile \
--add-exports=java.base/jdk.internal.reflect=org.graalvm.nativeimage.builder \
--add-exports=java.base/jdk.internal.util=org.graalvm.nativeimage.builder \
--add-exports=java.base/jdk.internal.vm.annotation=org.graalvm.nativeimage.builder \
--add-exports=java.base/jdk.internal.vm=org.graalvm.nativeimage.builder \
--add-exports=java.base/jdk.internal=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.invoke.util=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.net.www=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.net=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.nio.ch=org.graalvm.nativeimage.builder,org.graalvm.nativeimage.objectfile \
--add-exports=java.base/sun.reflect.annotation=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.reflect.generics.factory=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.reflect.generics.reflectiveObjects=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.reflect.generics.repository=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.reflect.generics.scope=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.reflect.generics.tree=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.security.jca=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.security.provider=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.security.rsa=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.security.ssl=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.security.util=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.security.x509=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.text.spi=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.util.calendar=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.util.cldr=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.util.locale.provider=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.util.locale=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.util.resources=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.util=org.graalvm.nativeimage.builder \
--add-exports=java.management/com.sun.jmx.mbeanserver=org.graalvm.nativeimage.builder \
--add-exports=java.management/sun.management=org.graalvm.nativeimage.builder,org.graalvm.nativeimage.pointsto \
--add-exports=jdk.graal.compiler/jdk.graal.compiler.util.json=com.oracle.graal.reporter \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.aarch64=com.oracle.graal.graal_enterprise,com.oracle.svm.svm_enterprise,jdk.graal.compiler,org.graalvm.nativeimage.builder,org.graalvm.nativeimage.objectfile \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.amd64=com.oracle.graal.graal_enterprise,com.oracle.svm.svm_enterprise,jdk.graal.compiler,org.graalvm.nativeimage.builder,org.graalvm.nativeimage.objectfile \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.code.site=com.oracle.svm.svm_enterprise,jdk.graal.compiler,org.graalvm.nativeimage.builder \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.code.stack=jdk.graal.compiler,org.graalvm.nativeimage.builder \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.code=com.oracle.graal.graal_enterprise,com.oracle.svm.svm_enterprise,jdk.graal.compiler,org.graalvm.nativeimage.builder,org.graalvm.nativeimage.objectfile,org.graalvm.nativeimage.pointsto,org.graalvm.truffle.compiler \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.common=com.oracle.graal.graal_enterprise,com.oracle.svm.svm_enterprise,jdk.graal.compiler,org.graalvm.nativeimage.builder,org.graalvm.nativeimage.pointsto \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.hotspot.aarch64=jdk.graal.compiler \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.hotspot.amd64=jdk.graal.compiler \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.hotspot.riscv64=jdk.graal.compiler \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.hotspot=com.oracle.graal.graal_enterprise,jdk.graal.compiler,org.graalvm.nativeimage.builder \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.meta=com.oracle.graal.graal_enterprise,com.oracle.svm.svm_enterprise,jdk.graal.compiler,org.graalvm.nativeimage.base,org.graalvm.nativeimage.builder,org.graalvm.nativeimage.configure,org.graalvm.nativeimage.objectfile,org.graalvm.nativeimage.pointsto,org.graalvm.truffle.compiler \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.riscv64=com.oracle.svm.svm_enterprise,jdk.graal.compiler,org.graalvm.nativeimage.builder \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.runtime=com.oracle.graal.graal_enterprise,jdk.graal.compiler,org.graalvm.nativeimage.builder,org.graalvm.nativeimage.pointsto \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.services=com.oracle.graal.graal_enterprise,com.oracle.svm.svm_enterprise,jdk.graal.compiler,org.graalvm.nativeimage.builder \
--add-exports=jdk.jfr/jdk.jfr.events=org.graalvm.nativeimage.builder \
--add-exports=jdk.jfr/jdk.jfr.internal.event=org.graalvm.nativeimage.builder \
--add-exports=jdk.jfr/jdk.jfr.internal.jfc=com.oracle.svm.svm_enterprise,org.graalvm.nativeimage.builder \
--add-exports=jdk.jfr/jdk.jfr.internal.settings=org.graalvm.nativeimage.builder \
--add-exports=jdk.jfr/jdk.jfr.internal.tracing=org.graalvm.nativeimage.builder \
--add-exports=jdk.jfr/jdk.jfr.internal=com.oracle.svm.svm_enterprise,org.graalvm.nativeimage.builder \
--add-exports=jdk.management/com.sun.management.internal=org.graalvm.nativeimage.builder \
-XX:+UseJVMCINativeLibrary \
-Xss10m \
-XX:+UseParallelGC \
-XX:MaxRAMPercentage=85.0 \
'-Dsvm.build.memoryUsageReasonText=$CI set to '"'"'true'"'"'' \
-XX:GCTimeRatio=9 \
-XX:+ExitOnOutOfMemoryError \
-Djava.awt.headless=true \
'-Dorg.graalvm.vendor=Oracle Corporation' \
-Dorg.graalvm.vendorurl=https://www.graalvm.org/ \
'-Dorg.graalvm.vendorversion=Oracle GraalVM 25.0.3+9.1' \
-Dorg.graalvm.version=25.0.3 \
-Dcom.oracle.graalvm.isaot=true \
-Djava.system.class.loader=com.oracle.svm.hosted.NativeImageSystemClassLoader \
-Dorg.graalvm.nativeimage.imagecode=buildtime \
-Xshare:off \
-Djdk.internal.lambda.disableEagerInitialization=true \
-Djdk.internal.lambda.eagerlyInitialize=false \
-Djava.lang.invoke.InnerClassLambdaMetafactory.initializeLambdas=false \
-Djava.lang.invoke.MethodHandle.DONT_INLINE_THRESHOLD=-1 \
-Djava.lang.invoke.MethodHandle.PROFILE_GWT=false \
-Djunit.platform.listeners.uid.tracking.output.dir=/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.grpc/grpc-opentelemetry/1.80.0/build/test-results/test/testlist \
--add-modules=ALL-DEFAULT \
--enable-native-access=org.graalvm.nativeimage.objectfile,org.graalvm.nativeimage.pointsto,org.graalvm.nativeimage.builder,org.graalvm.nativeimage.base,org.graalvm.nativeimage.configure,org.graalvm.nativeimage.foreign,com.oracle.svm.svm_enterprise,com.oracle.graal.reporter,org.graalvm.espresso.shared.svm,org.graalvm.nativeimage.shaded.capnproto \
--module-path \
/opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/builder/reporter.jar:/opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/builder/objectfile.jar:/opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/builder/svm-foreign.jar:/opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/builder/pointsto.jar:/opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/builder/native-image-base.jar:/opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/builder/espresso-svm.jar:/opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/builder/svm.jar:/opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/builder/svm-capnproto-runtime.jar:/opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/builder/svm-enterprise.jar:/opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/builder/svm-configure.jar \
-Dsvm.modulesupport.addedModules=java.rmi,java.sql,org.graalvm.collections,jdk.jdi,java.logging,java.xml.crypto,java.transaction.xa,java.xml,jdk.jfr,jdk.internal.vm.ci,java.datatransfer,org.graalvm.truffle.compiler,jdk.internal.md,java.desktop,java.naming,jdk.internal.ed,java.prefs,java.net.http,jdk.compiler,java.security.sasl,jdk.internal.opt,org.graalvm.nativeimage,jdk.zipfs,jdk.attach,java.base,jdk.management.agent,org.graalvm.word,jdk.internal.le,java.management,java.sql.rowset,jdk.unsupported,jdk.jlink,jdk.jdwp.agent,jdk.internal.jvmstat,java.instrument,java.security.jgss,jdk.management,java.compiler,java.scripting,jdk.graal.compiler,jdk.jdeps,java.management.rmi \
--add-modules=java.rmi,java.sql,org.graalvm.collections,jdk.jdi,java.logging,java.xml.crypto,java.transaction.xa,java.xml,jdk.jfr,jdk.internal.vm.ci,java.datatransfer,org.graalvm.truffle.compiler,jdk.internal.md,java.desktop,java.naming,jdk.internal.ed,java.prefs,java.net.http,jdk.compiler,java.security.sasl,jdk.internal.opt,org.graalvm.nativeimage,jdk.zipfs,jdk.attach,java.base,jdk.management.agent,org.graalvm.word,jdk.internal.le,java.management,java.sql.rowset,jdk.unsupported,jdk.jlink,jdk.jdwp.agent,jdk.internal.jvmstat,java.instrument,java.security.jgss,jdk.management,java.compiler,java.scripting,jdk.graal.compiler,jdk.jdeps,java.management.rmi \
--module \
org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner \
-keepalive \
/proc/10244/comm \
-imagecp \
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.grpc/grpc-opentelemetry/1.80.0/build/libs/io.grpc.grpc-opentelemetry_tests.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.graalvm.buildtools/junit-platform-native/1.0.0/2974f834a2f0a0105f2284fba7ef16dc4d60de56/junit-platform-native-1.0.0.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.jupiter/junit-jupiter/5.9.2/26c586fbe0ebd81b48c9f11f0d998124248697ae/junit-jupiter-5.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.platform/junit-platform-reporting/1.9.2/8bea292e2a546e4c21071f6bee60917ffd9f0101/junit-platform-reporting-1.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.platform/junit-platform-launcher/1.9.2/38e0bfad8c57d4cd1a8f27926c25ffe9543068d6/junit-platform-launcher-1.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.vintage/junit-vintage-engine/5.9.2/53421816bde124a564a64ba005dcc0c8e66a9722/junit-vintage-engine-5.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.jupiter/junit-jupiter-params/5.9.2/bc2765afb7b85b583c710dd259a11c6b8c39e912/junit-jupiter-params-5.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.jupiter/junit-jupiter-engine/5.9.2/572f7a553b53f83ee59cc045ce1c3772864ab76c/junit-jupiter-engine-5.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.jupiter/junit-jupiter-api/5.9.2/fed843581520eac594bc36bb4b0f55e7b947dda9/junit-jupiter-api-5.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.platform/junit-platform-engine/1.9.2/40aeef2be7b04f96bb91e8b054affc28b7c7c935/junit-platform-engine-1.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.platform/junit-platform-commons/1.9.2/6f9f8621d8230cd38aa42e58ccbc0c00569131ce/junit-platform-commons-1.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.platform/junit-platform-console/1.9.2/5548148dd55e820c42bd817ab93c1eaa05ce37f6/junit-platform-console-1.9.2.jar:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/tck-test-support/build/libs/tck-test-support-1.0.0-SNAPSHOT.jar:/home/runner/.gradle/caches/modules-2/files-2.1/io.grpc/grpc-opentelemetry/1.80.0/5459b7ca3aeb17b8d713f381cda33d4e6399c095/grpc-opentelemetry-1.80.0.jar:/home/runner/.gradle/caches/modules-2/files-2.1/io.grpc/grpc-inprocess/1.80.0/51b7c00db7070fcbc7b468533fb591d54385e327/grpc-inprocess-1.80.0.jar:/home/runner/.gradle/caches/modules-2/files-2.1/io.grpc/grpc-stub/1.80.0/ec14c383fd33981790afd29b4c3da231035087b0/grpc-stub-1.80.0.jar:/home/runner/.gradle/caches/modules-2/files-2.1/io.opentelemetry/opentelemetry-api/1.57.0/165669e71f583d1850bca050d5afe9f18e722b2e/opentelemetry-api-1.57.0.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.assertj/assertj-core/3.22.0/c300c0c6a24559f35fa0bd3a5472dc1edcd0111e/assertj-core-3.22.0.jar:/home/runner/.gradle/caches/modules-2/files-2.1/io.grpc/grpc-core/1.80.0/aae1d22e5a816650e1e7408f2f766bf769de76f1/grpc-core-1.80.0.jar:/home/runner/.gradle/caches/modules-2/files-2.1/io.grpc/grpc-context/1.80.0/b3fd476c0303c84ee5abe9a639f13020660f13b0/grpc-context-1.80.0.jar:/home/runner/.gradle/caches/modules-2/files-2.1/io.grpc/grpc-api/1.80.0/55f7b74a476f7f1815cb823b1a56b7ffd68e5043/grpc-api-1.80.0.jar:/home/runner/.gradle/caches/modules-2/files-2.1/com.google.guava/guava/33.5.0-android/8699de25f2f979108d6c1b804a7ba38cda1116bc/guava-33.5.0-jre.jar:/home/runner/.gradle/caches/modules-2/files-2.1/com.google.auto.value/auto-value-annotations/1.11.0/f0d047931d07cfbc6fa4079854f181ff62891d6f/auto-value-annotations-1.11.0.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.codehaus.mojo/animal-sniffer-annotations/1.26/2411ca884d78494fa6f1a5e614237305caf4be9/animal-sniffer-annotations-1.26.jar:/home/runner/.gradle/caches/modules-2/files-2.1/com.google.code.gson/gson/2.12.1/4e773a317740b83b43cfc3d652962856041697cb/gson-2.12.1.jar:/home/runner/.gradle/caches/modules-2/files-2.1/com.google.errorprone/error_prone_annotations/2.45.0/e4b613b411b934eee185aad72bee4c6148fb8dc3/error_prone_annotations-2.45.0.jar:/home/runner/.gradle/caches/modules-2/files-2.1/io.opentelemetry/opentelemetry-context/1.57.0/3e9212282c7b886d341cfc57ea116c24e9a2539a/opentelemetry-context-1.57.0.jar:/home/runner/.gradle/caches/modules-2/files-2.1/junit/junit/4.13.2/8ac9e16d933b6fb43bc7f576336b8f4d7eb5ba12/junit-4.13.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/com.google.code.findbugs/jsr305/3.0.2/25ea2e8b0c338a877313bd4672d3fe056ea78f0d/jsr305-3.0.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/com.google.guava/failureaccess/1.0.3/aeaffd00d57023a2c947393ed251f0354f0985fc/failureaccess-1.0.3.jar:/home/runner/.gradle/caches/modules-2/files-2.1/com.google.guava/listenablefuture/9999.0-empty-to-avoid-conflict-with-guava/b421526c5f297295adef1c886e5246c39d4ac629/listenablefuture-9999.0-empty-to-avoid-conflict-with-guava.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.jspecify/jspecify/1.0.0/7425a601c1c7ec76645a78d22b8c6a627edee507/jspecify-1.0.0.jar:/home/runner/.gradle/caches/modules-2/files-2.1/com.google.j2objc/j2objc-annotations/3.1/a892ca9507839bbdb900d64310ac98256cab992f/j2objc-annotations-3.1.jar:/home/runner/.gradle/caches/modules-2/files-2.1/com.google.android/annotations/4.1.1.4/a1678ba907bf92691d879fef34e1a187038f9259/annotations-4.1.1.4.jar:/home/runner/.gradle/caches/modules-2/files-2.1/io.perfmark/perfmark-api/0.27.0/f86f575a41b091786a4b027cd9c0c1d2e3fc1c01/perfmark-api-0.27.0.jar:/home/runner/.gradle/caches/modules-2/files-2.1/io.opentelemetry/opentelemetry-common/1.57.0/8f4cd60ad6255d57b397878a5af835e81d9693c8/opentelemetry-common-1.57.0.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.opentest4j/opentest4j/1.2.0/28c11eb91f9b6d8e200631d46e20a7f407f2a046/opentest4j-1.2.0.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.hamcrest/hamcrest-core/1.3/42a25dc3219429f0e5d060061f71acb49bf010a0/hamcrest-core-1.3.jar:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.grpc/grpc-opentelemetry/1.80.0/build/classes/java/test:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.grpc/grpc-opentelemetry/1.80.0/build/resources/test:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.grpc/grpc-opentelemetry/1.80.0/build/test-results/test/testlist \
-imagemp \
/opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/library-support.jar \
-H:CLibraryPath=/opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/clibraries/linux-amd64/glibc,/opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/clibraries/linux-amd64,/opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/clibraries \
-H:Path@driver=/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.grpc/grpc-opentelemetry/1.80.0/build/native/nativeTestCompile \
-H:BuildOutputJSONFile@user=/tmp/native-image-build-output.json \
'-H:ClassInitialization@jar:file:///home/runner/.gradle/caches/modules-2/files-2.1/io.opentelemetry/opentelemetry-api/1.57.0/165669e71f583d1850bca050d5afe9f18e722b2e/opentelemetry-api-1.57.0.jar!/META-INF/native-image/io.opentelemetry/opentelemetry-api/native-image.properties+api=io.opentelemetry.api.common.AttributeType:build_time' \
'-H:ResourceConfigurationResources@jar:file:///home/runner/.gradle/caches/modules-2/files-2.1/io.opentelemetry/opentelemetry-api/1.57.0/165669e71f583d1850bca050d5afe9f18e722b2e/opentelemetry-api-1.57.0.jar!/META-INF/native-image/io.opentelemetry/opentelemetry-api/resource-config.json+api=META-INF/native-image/io.opentelemetry/opentelemetry-api/resource-config.json' \
-H:FallbackThreshold@user+api=0 \
-H:Name@user+api=io.grpc.grpc-opentelemetry_tests-tests \
-H:ConfigurationFileDirectories@user=/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.grpc/grpc-opentelemetry/1.80.0/build/native/generated/generateTestResourcesConfigFile,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.opentest4j/opentest4j/1.2.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-commons/1.8.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.jupiter/junit-jupiter-params/5.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.jupiter/junit-jupiter-engine/5.8.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-engine/1.10.4,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-launcher/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-reporting/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.vintage/junit-vintage-engine/5.9.1,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/junit/junit/4.13.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.hamcrest/hamcrest-core/1.3,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.grpc/grpc-opentelemetry/1.80.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.grpc/grpc-api/1.27.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.google.code.findbugs/jsr305/3.0.1,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.google.errorprone/error_prone_annotations/2.6.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.google.guava/guava/33.6.0-jre,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.google.guava/failureaccess/1.0.1,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.google.guava/listenablefuture/9999.0-empty-to-avoid-conflict-with-guava,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.jspecify/jspecify/1.0.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.google.j2objc/j2objc-annotations/1.3,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.grpc/grpc-core/1.69.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.google.code.gson/gson/2.2.4,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.google.android/annotations/4.1.1.4,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.codehaus.mojo/animal-sniffer-annotations/1.23,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.perfmark/perfmark-api/0.25.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.grpc/grpc-context/1.57.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.opentelemetry/opentelemetry-api/1.19.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.opentelemetry/opentelemetry-context/1.19.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.opentelemetry/opentelemetry-common/1.56.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.google.auto.value/auto-value-annotations/1.10,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.grpc/grpc-inprocess/1.68.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.grpc/grpc-stub/1.70.0 \
-H:Optimize@user+api=b \
-H:+ReportExceptionStackTraces@user \
-H:Features@user+api=org.graalvm.junit.platform.JUnitPlatformFeature \
'-H:Class@explicit main-class=org.graalvm.junit.platform.NativeImageJUnitLauncher' \
-H:ImageBuildID@driver=e21ff390-6ff5-8481-6b4e-4f3b1e381648 \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/thirdparty/native-image.properties+api=com.oracle.svm.thirdparty.gson.GsonFeature' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties+api=com.oracle.svm.polyglot.groovy.GroovyIndyInterfaceFeature,com.oracle.svm.polyglot.scala.ScalaFeature'
]
========================================================================================================================
GraalVM Native Image: Generating 'io.grpc.grpc-opentelemetry_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                    (6.5s @ 0.17GB)
 Java version: 25.0.3+9-LTS, vendor version: Oracle GraalVM 25.0.3+9.1
 Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 C compiler: gcc (linux, x86_64, 13.3.0)
 Garbage collector: Serial GC (max heap size: 80% of RAM)
 2 user-specific feature(s):
 - com.oracle.svm.thirdparty.gson.GsonFeature
 - org.graalvm.junit.platform.JUnitPlatformFeature
------------------------------------------------------------------------------------------------------------------------
 1 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 12.67GB of memory (75.6% of system memory, $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.grpc/grpc-opentelemetry/1.80.0/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                   (39.4s @ 0.97GB)
    9,656 types,  11,763 fields, and  43,669 methods found reachable
    3,305 types,     514 fields, and   3,890 methods registered for reflection
       63 types,      71 fields, and      55 methods registered for JNI access
        0 downcalls and 0 upcalls registered for foreign access
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                               (5.4s @ 0.89GB)
[4/8] Parsing methods...      [**]                                                                       (4.0s @ 0.97GB)
[5/8] Inlining methods...     [***]                                                                      (2.4s @ 1.03GB)
[6/8] Compiling methods...    [*****]                                                                   (21.5s @ 0.77GB)
[7/8] Laying out methods...   [**]                                                                       (3.1s @ 0.98GB)
[8/8] Creating image...       [**]                                                                       (3.7s @ 1.12GB)
  11.56MB (37.32%) for code area:    27,006 compilation units
  15.01MB (48.44%) for image heap:  234,845 objects and 155 resources
   4.41MB (14.24%) for other data
  30.98MB in total image size, 28.58MB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   5.17MB java.base                                            3.41MB byte[] for code metadata
   2.43MB svm.jar (Native Image)                               2.89MB byte[] for java.lang.String
 569.31kB assertj-core-3.22.0.jar                              1.52MB java.lang.String
 483.05kB java.xml                                             1.21MB java.lang.Class
 412.41kB grpc-core-1.80.0.jar                                 1.00MB com.oracle.svm.core.hub.DynamicHubCompanion
 235.98kB java.rmi                                           709.61kB byte[] for general heap data
 199.14kB junit-jupiter-engine-5.9.2.jar                     456.10kB java.util.concurrent.ConcurrentHashMap$Node
 164.41kB grpc-api-1.80.0.jar                                369.26kB byte[] for reflection metadata
 123.12kB junit-platform-launcher-1.9.2.jar                  357.73kB java.util.HashMap$Node
 120.44kB guava-33.5.0-jre.jar                               290.63kB java.lang.Object[]
   1.23MB for 41 more packages                                 2.79MB for 2161 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 33 component(s) is embedded in binary (1.47kB). 5 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 FUTR: Use '--future-defaults=all' to prepare for future releases.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                        5.9s (6.6% of total time) in 843 GCs | Peak RSS: 1.91GB | CPU load: 3.63
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.grpc/grpc-opentelemetry/1.80.0/build/native/nativeTestCompile/io.grpc.grpc-opentelemetry_tests-tests (executable)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.grpc/grpc-opentelemetry/1.80.0/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library)
 /tmp/native-image-build-output.json (build_info)
========================================================================================================================
Finished generating 'io.grpc.grpc-opentelemetry_tests-tests' in 1m 28s.
[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.grpc/grpc-opentelemetry/1.80.0/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.grpc/grpc-opentelemetry/1.80.0/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.grpc/grpc-opentelemetry/1.80.0/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report

FAILURE: ----------------------------------------

io_grpc.grpc_opentelemetry.Grpc_opentelemetryTest > grpcTraceBinPropagatorInjectsAndExtractsSampledTextCarrier() FAILED

Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
io_grpc.grpc_opentelemetry.Grpc_opentelemetryTest > grpcTraceBinPropagatorAdvertisesSingletonField() SUCCESSFUL

> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.grpc/grpc-opentelemetry/1.80.0/build/native/nativeTestCompile/io.grpc.grpc-opentelemetry_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 1m 32s
io_grpc.grpc_opentelemetry.Grpc_opentelemetryTest > grpcOpenTelemetryRegisterGlobalAppliesToNewBuilders() SUCCESSFUL

io_grpc.grpc_opentelemetry.Grpc_opentelemetryTest > grpcTraceBinPropagatorPreservesUnsampledTraceFlags() SUCCESSFUL

io_grpc.grpc_opentelemetry.Grpc_opentelemetryTest > grpcOpenTelemetryConfiguresInProcessClientAndServerBuilders() SUCCESSFUL

io_grpc.grpc_opentelemetry.Grpc_opentelemetryTest > grpcTraceBinPropagatorIgnoresNullInvalidAndMalformedInputs() SUCCESSFUL

io_grpc.grpc_opentelemetry.Grpc_opentelemetryTest > grpcOpenTelemetryUsesExplicitOpenTelemetrySdkForInstrumentation() SUCCESSFUL

io_grpc.grpc_opentelemetry.Grpc_opentelemetryTest > grpcOpenTelemetryDoesNotMaskRpcFailures() SUCCESSFUL


Failures (1):
  JUnit Jupiter:Grpc_opentelemetryTest:grpcTraceBinPropagatorInjectsAndExtractsSampledTextCarrier()
    MethodSource [className = 'io_grpc.grpc_opentelemetry.Grpc_opentelemetryTest', methodName = 'grpcTraceBinPropagatorInjectsAndExtractsSampledTextCarrier', methodParameterTypes = '']
    => org.graalvm.nativeimage.MissingReflectionRegistrationError: Cannot reflectively invoke method 'public java.lang.Object java.util.HashMap.clone()'. To allow this operation, add the following to the 'reflection' section of 'reachability-metadata.json' and rebuild the native image:

  {
    "type": "java.util.HashMap",
    "methods": [
      {
        "name": "clone",
        "parameterTypes": []
      }
    ]
  }

The 'reachability-metadata.json' file should be located in 'META-INF/native-image/<group-id>/<artifact-id>/' of your project. For further help, see https://www.graalvm.org/latest/reference-manual/native-image/metadata/#reflection
       org.graalvm.nativeimage.builder/com.oracle.svm.core.reflect.MissingReflectionRegistrationUtils.reportInvokedExecutable(MissingReflectionRegistrationUtils.java:110)
       java.base@25.0.3/java.lang.reflect.Method.acquireMethodAccessor(Method.java:149)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:562)
       org.assertj.core.internal.Maps.clone(Maps.java:424)
       org.assertj.core.internal.Maps.getNotExpectedKeys(Maps.java:405)
       org.assertj.core.internal.Maps.assertContainsOnlyKeys(Maps.java:369)
       org.assertj.core.internal.Maps.assertContainsOnlyKeys(Maps.java:353)
       org.assertj.core.api.AbstractMapAssert.containsOnlyKeysForProxy(AbstractMapAssert.java:1174)
       org.assertj.core.api.AbstractMapAssert.containsOnlyKeys(AbstractMapAssert.java:1167)
       io_grpc.grpc_opentelemetry.Grpc_opentelemetryTest.grpcTraceBinPropagatorInjectsAndExtractsSampledTextCarrier(Grpc_opentelemetryTest.java:96)
       [...]

Test run finished after 6 ms
[         3 containers found      ]
[         0 containers skipped    ]
[         3 containers started    ]
[         0 containers aborted    ]
[         3 containers successful ]
[         0 containers failed     ]
[         8 tests found           ]
[         0 tests skipped         ]
[         8 tests started         ]
[         0 tests aborted         ]
[         7 tests successful      ]
[         1 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.grpc/grpc-opentelemetry/1.80.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
17 actionable tasks: 7 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for io.grpc:grpc-opentelemetry:1.80.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 33s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 1.80.0 at 2026-05-12T10:22:09Z with exit code 1 after 94s
❌ ERROR: 'native-image run' for 1.80.0 failed with exit code 1.
FAILED[native-image run][1.80.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.80.0" ./gradlew clean test -Pcoordinates="io.grpc:grpc-opentelemetry:1.80.0"]
```

--- a/tests/src/io.grpc/grpc-opentelemetry/1.80.0/src/test/java/io_grpc/grpc_opentelemetry/Grpc_opentelemetryTest.java
+++ b/tests/src/io.grpc/grpc-opentelemetry/1.80.0/src/test/java/io_grpc/grpc_opentelemetry/Grpc_opentelemetryTest.java
@@ -93,7 +93,7 @@ public class Grpc_opentelemetryTest {
 
         propagator.inject(originalContext, carrier, MAP_SETTER);
 
-        assertThat(carrier).containsOnlyKeys(GrpcTraceBinContextPropagator.GRPC_TRACE_BIN_HEADER);
+        assertThat(carrier.keySet()).containsExactly(GrpcTraceBinContextPropagator.GRPC_TRACE_BIN_HEADER);
         assertThat(carrier.get(GrpcTraceBinContextPropagator.GRPC_TRACE_BIN_HEADER))
                 .isNotBlank()
                 .doesNotContain("=");
@@ -132,7 +132,7 @@ public class Grpc_opentelemetryTest {
         propagator.inject(null, carrier, MAP_SETTER);
         propagator.inject(Context.root().with(Span.wrap(SpanContext.getInvalid())), carrier, MAP_SETTER);
 
-        assertThat(carrier).isEmpty();
+        assertThat(carrier.keySet()).isEmpty();
         assertThat(propagator.extract(null, carrier, MAP_GETTER)).isSameAs(Context.root());
 
         Context existingContext = Context.root().with(Span.wrap(sampledSpanContext()));

